File: Classes/WMISet/WMISet/Examples/cbuilder/RegEditor/FrmGetNameU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FrmGetNameU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFrmGetName *FrmGetName;
//---------------------------------------------------------------------------
__fastcall TFrmGetName::TFrmGetName(AnsiString APrompt, AnsiString AValue)
        : TForm((TComponent *) NULL)
{
  lblPrompt->Caption = APrompt;
  Caption = APrompt;
  edtName->Text = AValue;
}
//---------------------------------------------------------------------------

File: Classes/WMISet/WMISet/Examples/delphi/RegEditor/FrmGetNameU.pas
unit FrmGetNameU;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls;

type
  TFrmGetName = class(TForm)
    lblPrompt: TLabel;
    edtName: TEdit;
    btnOk: TButton;
    Cancel: TButton;
  private
    { Private declarations }
  public
    { Public declarations }
    constructor Create(APrompt, AValue: string); reintroduce;
  end;

var
  FrmGetName: TFrmGetName;

implementation

{$R *.DFM}

{ TForm1 }

constructor TFrmGetName.Create(APrompt, AValue: string);
begin
  inherited Create(nil);
  lblPrompt.Caption := APrompt;
  Caption := APrompt;
  edtName.Text := AValue;
end;

end.
